sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PTransform.java

```diff
@@ -178,6 +179,34 @@ public String getName() {
     return name != null ? name : getKindString();
   }
 
+  /**
+   * Sets resource hints for the transform.
+   *
+   * @param resourceHints a {@link ResourceHints} instance.
+   * @return a reference to the same transfrom instance.
+   *     <p>For example:
+   *     <pre>{@code
+   * Pipeline p = ...
+   * ...
+   * p.apply(new SomeTransform().setResourceHints(ResourceHints.create().withMinRam("6 GiB")))
+   * ...
+   *
+   * }</pre>
+   */
+  public PTransform<InputT, OutputT> setResourceHints(ResourceHints resourceHints) {
+    this.resourceHints = resourceHints;
+    return this;
+  }
+
+  /** Returns resource hints set on the transform. */
+  public ResourceHints getResourceHints() {
+    if (resourceHints == null) {
```

Comment:
Is this preferable to initializing it in the constructor (or at field declaration)?

Reply:
I think we can make it field declaration. We needed this pattern in Python since Python (unlike Java) does not call superclass constructor by default.